docs/source/developer_guide/index.rst

```diff
@@ -4,80 +4,154 @@
+It is vital to make sure other people can understand your changes. When
+adding a new module or feature to the PySyft library, we encourage you
+to ensure code quality by documenting your code.
 
-Issue Allocation
-^^^^^^^^^^^^^^^^
+We ask you to follow the appropriate style guide to its best practices
+for the language you’re working in for documentation and general code
+cleanliness.
 
-Each issue someone is currently working on should have an assignee. If you want to contribute to an issue someone else is already working on please make sure to get in contact with that person via Slack or Github and organize yourself.
+OpenMined enforces the following style guides by language:
 
-If you want to work on an open issue, please post a comment on that issue and we will assign you as the assignee.
+-  `Python <https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html>`__
+-  `Javascript <https://prettier.io/>`__
+-  `C++ <https://google.github.io/styleguide/cppguide.html>`__
+-  `Java <https://google.github.io/styleguide/javaguide.html>`__
+-  `Kotlin <https://kotlinlang.org/docs/coding-conventions.html>`__
+-  `Swift <https://google.github.io/swift/>`__
+-  `Go <https://go.dev/doc/effective_go>`__
+-  `Rust <https://doc.rust-lang.org/1.0.0/style/README.html>`__
+-  `Julia <https://docs.julialang.org/en/v1/manual/style-guide/>`__
+-  `R <https://google.github.io/styleguide/Rguide.html>`__
 
-.. note:: We try our best to keep the assignee up-to-date, but as we are all humans with our own scheduling issues, make sure to check the comments once before you start working on an issue even when no one is assigned to it.
+..
 
-Writing Test Cases
-^^^^^^^^^^^^^^^^^^
+   Note: As with any software project, keeping the amount of code to a
+   minimum is essential, so keep code duplication to a minimum!
 
-Always make sure to create the necessary tests and keep test coverage as high or higher than it was before you. You can always ask for help in Slack or Github if you don't feel confident about your tests. We're happy to help!
+5. Pull Requests
+~~~~~~~~~~~~~~~~
 
-Documentation and Codestyle
-^^^^^^^^^^^^^^^^^^^^^^^^^^^
+At any point in time, you can send a `GitHub Pull
+Request <https://github.com/OpenMined/PySyft/pulls>`__ to PySyft (read
+more about `pull
+requests <https://docs.github.com/en/pull-requests>`__), so others can
+see your changes and give you feedback.
 
-To ensure code quality and make sure other people can understand your changes, you have to document your code. For documentation and general code cleanliness, we ask that you `follow the appropriate styleguide <https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md>`_ for the language you're working in.
+Before sending a PR, it is crucial to make sure you comply with the
+below instructions:
 
-Keep it DRY (Don't repeat yourself)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+-  No commits should be made to the ``master`` branch directly.
```

Comment:
Since our default branch is "dev", prolly we can change this? What do you think?

Reply:
Makes more sense , thanks ☺️

Reply:
We can mention both :)